core/src/main/java/io/substrait/relation/RelProtoConverter.java

```diff
@@ -75,6 +77,43 @@ private List<SortField> toProtoS(Collection<Expression.SortField> sorts) {
         .collect(java.util.stream.Collectors.toList());
   }
 
+  private List<ConsistentPartitionWindowRel.WindowRelFunction> toProtoWindowRelFunctions(
+      Collection<ConsistentPartitionWindow.WindowRelFunctionInvocation>
+          windowRelFunctionInvocations) {
+
+    return windowRelFunctionInvocations.stream()
+        .map(
+            f -> {
+              var argVisitor = FunctionArg.toProto(typeProtoConverter, exprProtoConverter);
+              var args = f.arguments();
+              var aggFuncDef = f.declaration();
+
+              var arguments =
+                  IntStream.range(0, args.size())
+                      .mapToObj(i -> args.get(i).accept(aggFuncDef, i, argVisitor))
+                      .collect(Collectors.toList());
+              var options =
+                  f.options().entrySet().stream()
+                      .map(
+                          o ->
+                              FunctionOption.newBuilder()
+                                  .setName(o.getKey())
+                                  .addAllPreference(o.getValue().values())
+                                  .build())
+                      .collect(java.util.stream.Collectors.toList());
+
+              return ConsistentPartitionWindowRel.WindowRelFunction.newBuilder()
+                  .setInvocation(f.invocation().toProto())
+                  .setPhase(f.aggregationPhase().toProto())
+                  .setOutputType(toProto(f.outputType()))
+                  .addAllArguments(arguments)
+                  .addAllOptions(options)
+                  .setFunctionReference(functionCollector.getFunctionReference(f.declaration()))
```

Comment:
I think we need to set the `lowerBound`, `upperBound` and `boundsType` as well.

Reply:
Very good catch! 💯